test/test/nonjava/non_java_app.cpp

```diff
@@ -0,0 +1,283 @@
+    void* lib = dlopen(profiler_lib_path, RTLD_NOW | RTLD_GLOBAL);
+    if (lib == NULL) {
+        std::cerr << dlerror() << std::endl;
+        exit(1);
+    }
+
+    _asprof_init = (asprof_init_t)dlsym(lib, "asprof_init");
+    _asprof_execute = (asprof_execute_t)dlsym(lib, "asprof_execute");
+    _asprof_error_str = (asprof_error_str_t)dlsym(lib, "asprof_error_str");
+
+    _asprof_init();
+}
+
+void startProfiler() {
+    asprof_error_t err = _asprof_execute("start,event=cpu,interval=1ms,cstack=dwarf", outputCallback);
+    if (err != NULL) {
+        std::cerr << _asprof_error_str(err) << std::endl;
+        exit(1);
+    }
+}
+
+void stopProfiler(char* outputFile) {
+    char cmd[4096];
+    snprintf(cmd, sizeof(cmd), "stop,file=%s", outputFile);
+    asprof_error_t err = _asprof_execute(cmd, outputCallback);
+    if (err != NULL) {
+        std::cerr << _asprof_error_str(err) << std::endl;
+        exit(1);
+    }
+}
+
+void loadJvmLib() {
+    char* java_home = getenv("JAVA_HOME");
+    if (java_home == NULL) {
+        std::cerr << "JAVA_HOME is not set" << std::endl;
+        exit(1);
+    }
+    char lib_path[4096];
+    snprintf(lib_path, sizeof(lib_path), "%s/%s", java_home, jvm_lib_path);
+    _jvm_lib = dlopen(lib_path, RTLD_LOCAL | RTLD_NOW);
+    if (_jvm_lib == NULL) {
+        std::cerr << "Unable to find: " << lib_path << ", Error: " << dlerror() << std::endl;
+        exit(1);
+    }
+}
+
+void startJvm() {
+    // Start JVM
+    JavaVMInitArgs vm_args;
+    JavaVMOption options[2];
+    int pid = getpid();
+
+    options[0].optionString = const_cast<char*>("-Djava.class.path=build/test");
+    options[1].optionString = const_cast<char*>("-Xcheck:jni");
+
+    // Configure JVM
+    vm_args.version = JNI_VERSION_10;
+    vm_args.nOptions = 2;
+    vm_args.options = options;
+    vm_args.ignoreUnrecognized = true;
+
+    CreateJvm createJvm = (CreateJvm)dlsym(_jvm_lib, "JNI_CreateJavaVM");
+    if (createJvm == NULL) {
+        std::cerr << "Unable to find: JNI_CreateJavaVM" << std::endl;
+        exit(1);
+    }
+
+    // Create the JVM
+    jint rc = createJvm(&_jvm, (void**)&_env, &vm_args);
+    if (rc != JNI_OK) {
+        std::cerr << "Failed to create JVM" << std::endl;
+        exit(1);
+    }
+}
+
+void executeJvmTask() {
+    jclass customClass = _env->FindClass("JavaClass");
+    if (customClass == nullptr) {
+        std::cerr << "Can't find JavaClass" << std::endl;
+        exit(1);
+    }
+
+    jmethodID cpuHeavyTask = _env->GetStaticMethodID(customClass, "cpuHeavyTask", "()D");
+    if (cpuHeavyTask == nullptr) {
+        std::cerr << "Can't find cpuHeavyTask" << std::endl;
+        exit(1);
+    }
+
+    for (int i = 0; i < 100; ++i) {
+        jdouble result = _env->CallStaticDoubleMethod(customClass, cpuHeavyTask);
+        if (_env->ExceptionCheck()) {
+            jthrowable exception = _env->ExceptionOccurred();
+            _env->ExceptionDescribe();
+            _env->ExceptionClear();
```

Comment:
You've listed all possible JNI exception functions :)
`ExceptionDescribe` is enough. It does everything: prints exception message with a stack trace and clears current exception.